src/containers/ContributorsSectionFromPkg.tsx

```diff
@@ -0,0 +1,116 @@
+/* @jsx MD */
+import MD, { awaitComponent, Component } from "jsx-md";
+import parseLinkHeader from "parse-link-header";
+
+import { Octokit } from "@octokit/core";
+
+import { ContributorsSection } from "../components";
+import { extractGithubOwnerAndRepo } from "../components/badges/utils/extractGithubOwnerAndRepo";
+import { PackageJSON } from "../PackageJSON";
+
+import type { Contributor } from "../components/ContributorsList";
+
+// Fetch total contributors count using pagination trick. See https://stackoverflow.com/a/44347632.
+const fetchTotalContributorsCount = async (
+  client: Octokit,
+  owner: string,
+  repo: string
+) => {
+  const resp = await client.request(
+    "GET /repos/{owner}/{repo}/contributors{?per_page}",
+    { owner, repo, per_page: 1 }
+  );
+  if (!resp.headers.link) {
+    return 0;
+  }
+  const links = parseLinkHeader(resp.headers.link);
+  if (!links) {
+    return 0;
+  }
+  return parseInt(links.last?.page);
+};
+
+// List repository contributors using v3 GitHub API, See https://docs.github.com/en/rest/reference/repos#list-repository-contributors.
+const fetchContributors = async (
+  client: Octokit,
+  owner: string,
+  repo: string,
+  fetchUntil: number
+) => {
+  let contributors: Array<Contributor> = [];
+  for (let page = 1; contributors.length < fetchUntil; page++) {
+    const resp = await client.request(
+      "GET /repos/{owner}/{repo}/contributors",
+      { owner, repo, page }
+    );
+    // No more contributors.
+    if (resp.data.length <= 0) {
+      return contributors;
+    }
+    contributors = contributors.concat(
+      resp.data
+        // NOTE: Exclude "Bot" and repo owner as contributors does'nt work as total contributors count will include them as well.
+        // .filter(
+        //   (contributor) =>
+        //     contributor.type !== "Bot" && contributor.login !== owner
+        // )
```

Comment:
Not sure why we can't filter out bots:

```suggestion.filter((contributor) => contributor.type !== "Bot")
```

This way, bots should now be in the `additionalContributorsCount`, if I understood the code correctly

Reply:
Yes you are right, previous I thought that the total contributors displaying does not match the `contributorsMaxDisplayLength` prop provided. Looks like it is working as expected because Bots and owner will be added into the additional contributors count instead.